extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/HealthOpenAPIFilter.java

```diff
@@ -0,0 +1,214 @@
+
+        // Readiness
+        paths.addPathItem(readinessPath, createReadinessPathItem());
+    }
+
+    private PathItem createHealthPathItem() {
+        PathItem pathItem = new PathItemImpl();
+        pathItem.setDescription("MicroProfile Health Endpoint");
+        pathItem.setSummary(
+                "MicroProfile Health provides a way for your application to distribute information about its healthiness state to state whether or not it is able to function properly");
+        pathItem.setGET(createHealthOperation());
+        return pathItem;
+    }
+
+    private PathItem createLivenessPathItem() {
+        PathItem pathItem = new PathItemImpl();
+        pathItem.setDescription("MicroProfile Health - Liveness Endpoint");
+        pathItem.setSummary(
+                "Liveness checks are utilized to tell whether the application should be restarted");
+        pathItem.setGET(createLivenessOperation());
+        return pathItem;
+    }
+
+    private PathItem createReadinessPathItem() {
+        PathItem pathItem = new PathItemImpl();
+        pathItem.setDescription("MicroProfile Health - Readiness Endpoint");
+        pathItem.setSummary(
+                "Readiness checks are used to tell whether the application is able to process requests");
+        pathItem.setGET(createReadinessOperation());
+        return pathItem;
+    }
+
+    private Operation createHealthOperation() {
+        Operation operation = new OperationImpl();
+        operation.setDescription("Check the health of the application");
+        operation.setOperationId("microprofile_health_root");
+        operation.setTags(MICROPROFILE_HEALTH_TAG);
+        operation.setSummary("An aggregated view of the Liveness and Readiness of this application");
+        operation.setResponses(createAPIResponses());
+        return operation;
+    }
+
+    private Operation createLivenessOperation() {
+        Operation operation = new OperationImpl();
+        operation.setDescription("Check the liveness of the application");
+        operation.setOperationId("microprofile_health_liveness");
+        operation.setTags(MICROPROFILE_HEALTH_TAG);
+        operation.setSummary("The Liveness check of this application");
+        operation.setResponses(createAPIResponses());
+        return operation;
+    }
+
+    private Operation createReadinessOperation() {
+        Operation operation = new OperationImpl();
+        operation.setDescription("Check the readiness of the application");
+        operation.setOperationId("microprofile_health_readiness");
+        operation.setTags(MICROPROFILE_HEALTH_TAG);
+        operation.setSummary("The Readiness check of this application");
+        operation.setResponses(createAPIResponses());
+        return operation;
+    }
+
+    private APIResponses createAPIResponses() {
+        APIResponses responses = new APIResponsesImpl();
+        responses.addAPIResponse("200", createAPIResponse());
```

Comment:
should also have 503 and 500 - https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/protocol-wireformat.adoc#status-codes

Reply:
Thanks @xstefank - I'll add those.
W.r.t experimental paths, I am easy, I do not really have an opinion on that. Maybe @knutwannheden can comment here as this is requested by him.

Reply:
I don't understand what you mean by "experimental paths", but from my perspective it is only important that the path of every endpoint, which gets called by another service, is registered with the Azure API management. Having the API responses described is not necessary, but of course a nice to have!

Reply:
@knutwannheden sorry, I meant experimental **features** paths, e.g. `/health/group/{name}` or `/health/well`.